### PR TITLE
Remember last choice of download files in import dialog

### PR DESCRIPTION
**Pull request by tobiasdiez**
The choice to download linked files in the import dialog is now remembered, and automatically restored. So if you downloaded the files the last time, the option is selected on the next import. This made the corresponding option in the preference dialog obsolete.

Rest is a bit of code cleanup.

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/preferences/ImportTab.fxml [hunk 1]
```diff
@@ -28,8 +27,7 @@
                   prefWidth="300" minWidth="300" maxWidth="300"/>
 
         <Label text="%File directory pattern" GridPane.rowIndex="1"/>
-        <TextField fx:id="fileDirPattern" GridPane.columnIndex="1" GridPane.rowIndex="1"
+        <TextField fx:id="fileDirectoryPattern" GridPane.columnIndex="1" GridPane.rowIndex="1"
                    prefWidth="300" minWidth="300" maxWidth="300"/>
-        <CheckBox fx:id="downloadLinkedFiles" text="%Download linked online files" GridPane.rowIndex="2"/>
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/ImportTab.fxml:33]: Is there a reason you removed the "Download linked online files"`?

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/preferences/ImportTab.fxml:33]: Yes, there is no need for this checkbox in the preferences if its also displayed in the dialog anyway.

File: src/main/java/org/jabref/model/metadata/FilePreferences.java [hunk 2]
```diff
@@ -52,7 +52,11 @@ public String getFileDirPattern() {
...
 
-    public boolean getDownloadLinkedFiles() {
-        return downloadLinkedFiles;
+    public boolean shouldDownloadLinkedFiles() {
+        return shouldDownloadLinkedFiles;
+    }
+
+    public void setShouldDownloadLinkedFiles(boolean shouldDownloadLinkedFiles) {
+        this.shouldDownloadLinkedFiles = shouldDownloadLinkedFiles;
     }
```
Comment (calixtus) [inline src/main/java/org/jabref/model/metadata/FilePreferences.java:61]:
```suggestion
    public FilePreferences withShouldDownloadLinkedFiles(boolean shouldDownloadLinkedFiles) {
        return new FilePreferences(this.user,
                           this.mainFileDirectory,
                           this.bibLocationAsPrimary,
                           this.fileNamePattern,
                           this.fileDirPattern,
                           shouldDownloadLinkedFiles);
    }
```

File: src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java [hunk 3]
```diff
@@ -130,10 +131,15 @@ public void importEntries(List<BibEntry> entriesToImport, boolean downloadFiles)
             buildImportHandlerThenImportEntries(entriesToImport);
         }
 
-        if (downloadFiles) {
+        // Remember the selection in the dialog
+        FilePreferences filePreferences = preferences.getFilePreferences();
+        filePreferences.setShouldDownloadLinkedFiles(shouldDownloadFiles);
+        preferences.storeFilePreferences(filePreferences);
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java:137]:
```suggestion
        storeFilePreferences(filePreferences.withShouldDownloadLinkedFiles(shouldDownloadFiles));
```

**Comment by tobiasdiez** [comment]
What is our convention on immutable preference objects? I was a bit surprised to find that the variables in FilePreferences were final, I was expecting a normal POJO.

**Comment by calixtus** [comment]
I think we don't have any policy yet. I have just read in effective Java about minimizing mutability wherever possible (chap. 15 I think).

**Comment by Siedlerchr** [comment]
DevCall decision: Merge now and @calixtus  will rework that later.

**Pull request merged**